src/__tests__/ParseQuery-test.js

```diff
@@ -2709,6 +2709,53 @@ describe('ParseQuery', () => {
     });
   });
 
+  it('can issue a search query', async () => {
+    CoreManager.setQueryController({
+      find() {},
+      aggregate(className, params, options) {
+        expect(className).toBe('Item');
+        expect(params).toEqual({
+          $search: {
+            index: 'searchIndex',
+            text: {
+              path: ['name'],
+              query: 'searchTerm',
+            },
+          },
+        });
+        expect(options.useMasterKey).toEqual(true);
+        return Promise.resolve({
+          results: [],
+        });
+      },
+    });
+    const q = new ParseQuery('Item');
+    await q.search('searchTerm', ['name'], { index: 'searchIndex' });
+  });
+
+  it('search term is required', async () => {
+    const q = new ParseQuery('Item');
+    await expect(q.search()).rejects.toThrow('A search term is required.');
+  });
+
+  it('search term must be a string', async () => {
+    const q = new ParseQuery('Item');
+    await expect(q.search(123)).rejects.toThrow('The value being searched for must be a string.');
+  });
+
+  it('search can return an empty array if no results', async () => {
```

Comment:
Can you add an integration test? You can use those results to finish unit test coverage

Reply:
This feature only works with Mongodb [atlas](https://www.mongodb.com/docs/atlas/atlas-search/index-definitions/) search index, not sure it'll work with mongodb runner